### PR TITLE
Add Conflux Network support

### PR DESCRIPTION
**Pull request by aalu1418**
Hello!

We're in the process of getting Conflux Network supported with Chainlink. This is the external initiators portion of the integration. Please let me know if there are any necessary changes. The base of the code was the ETH code, but customized to meet Conflux RPC calls and parameters.

### CODE REVIEW

**Comment by boxhock** [comment]
Hi there, thanks for making this PR!

Is Conflux EVM based? In case we can use our solidity contracts on Conflux, I think it would make more sense to copy the BSC integration.
The reason for this is that the ETH integration here is very generic (as we already have a native ETH integration in cl-core), while the BSC integration is specifically made for our solidity contracts.
There's not a huge difference, so updating the PR shouldn't be too difficult.

Also, does Conflux support WS, or only HTTP RPC?

**Comment by aalu1418** [comment]
Hello!
Yes Conflux does use EVM with a few small tweaks (it uses epochs instead of block numbers). I will take a look at the BSC components. Are there specific things I should take a look at?

I think there is, but I am not sure how far along in development it is. I will find out, and add it to the initiator accordingly.

Thanks for the feedback!

**Comment by aalu1418** [comment]
@boxhock I took a look the BSC components, and I like how it uses the evm + eth components, but Conflux uses the EVM (so Solidity functions normally) with differences in the JSON-RPC method names. They are `cfx_*` instead of `eth_*`. And then instead of `blockNumber`, `toBlock`, etc we use `epochNumber`, `toEpoch` etc.

So I don't think I can base it off the BSC components without changing the `eth.go` components. Please let me know if you see a better way! (also still checking on the websocket)

**Comment by boxhock** [comment]
Hey, been a bit busy here but thanks for your patience,

I think the specific things you should take a look at is how we make the filter query: https://github.com/smartcontractkit/external-initiator/blob/master/blockchain/binance-smart-chain.go#L27
And how we parse log events to an oracle request: https://github.com/smartcontractkit/external-initiator/blob/master/blockchain/binance-smart-chain.go#L133

These functions will make it possible to work with our on-chain oracle contracts. Other than that, feel free to use your `cfx_*` functions instead.

Does this make sense?

**Comment by aalu1418** [comment]
makes sense! I will take a look and get back to you about it

**Comment by aalu1418** [comment]
included those two specific functions and all the test cases and integrations still work.

Still to-do: websocket integration

**Comment by aalu1418** [comment]
@boxhock I've added the WS functionality as well as the two smart contract specific functions. please let me know if there's anything else!

**Review by boxhock — CHANGES_REQUESTED** [review]
Just a few items to point out

File: blockchain/cfx.go [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+type cfxLogResponse struct {
+	LogIndex         string   			`json:"logIndex"`
+	EpochNumber      string   			`json:"epochNumber"`
+	BlockHash        common.Hash 		`json:"blockHash"`
+	TransactionHash  common.Hash 		`json:"transactionHash"`
+	TransactionIndex string   			`json:"transactionIndex"`
+	Address          common.Address `json:"address"`
+	Data             string   			`json:"data"`
+	Topics           []common.Hash 	`json:"topics"`
+}
```
[the remark this answers is not in the document]
Comment (aalu1418) [reply blockchain/cfx.go:150]: fixed! it looked right in my editor, but the tabs must do something strange

[the remark this answers is not in the document]
Comment (aalu1418) [reply blockchain/cfx.go:150]: fixed! it looked right in my editor, but the tabs did something strange on github

**Comment by boxhock** [comment]
@aalu1418 could you please rebase?

There seems there's also some issues with formatting. Check out the failing "golang/static / Imports" test. Most issues are solved by running go fmt.

**Comment by aalu1418** [comment]
@boxhock rebased, and fixed the formatting error/failing test. thanks for the tip!

**Review by boxhock — APPROVED** [review]
Looks good, thanks 👍

**Pull request merged**